commands/store.go

```diff
@@ -0,0 +1,245 @@
+	storeDeployCmd.Flags().StringArrayVar(&secrets, "secret", []string{}, "Give the function access to a secure secret")
+
+	// Set bash-completion.
+	_ = storeDeployCmd.Flags().SetAnnotation("handler", cobra.BashCompSubdirsInDir, []string{})
+
+	storeCmd.AddCommand(storeListCmd)
+	storeCmd.AddCommand(storeInspectCmd)
+	storeCmd.AddCommand(storeDeployCmd)
+	faasCmd.AddCommand(storeCmd)
+}
+
+var storeCmd = &cobra.Command{
+	Use:   `store`,
+	Short: "OpenFaaS store commands",
+	Long:  "Allows browsing and deploying OpenFaaS store functions",
+}
+
+var storeListCmd = &cobra.Command{
+	Use:     `list [--store STORE_URL]`,
+	Short:   "List OpenFaaS store items",
+	Long:    "Lists the available items in OpenFaas store",
+	Example: `  faas-cli store list --store https://domain:port`,
+	RunE:    runStoreList,
+}
+
+var storeInspectCmd = &cobra.Command{
+	Use:     `inspect (FUNCTION_NAME|FUNCTION_TITLE) [--store STORE_URL]`,
+	Short:   "Show OpenFaaS store function details",
+	Long:    "Prints the detailed informations of the specified OpenFaaS function",
+	Example: `  faas-cli store inspect NodeInfo --store https://domain:port`,
+	RunE:    runStoreInspect,
+}
+
+var storeDeployCmd = &cobra.Command{
+	Use: `deploy (FUNCTION_NAME|FUNCTION_TITLE)
+							[--lang <ruby|python|node|csharp>]
+							[--gateway GATEWAY_URL]
+							[--handler HANDLER_DIR]
+							[--env ENVVAR=VALUE ...]
+							[--label LABEL=VALUE ...]
+							[--replace=false]
+							[--update=false]
+							[--constraint PLACEMENT_CONSTRAINT ...]
+							[--regex "REGEX"]
+							[--filter "WILDCARD"]
+							[--secret "SECRET_NAME"]`,
+
+	Short: "Deploy OpenFaaS functions from the store",
+	Long:  `Same as faas-cli deploy except pre-loaded with arguments from the store`,
+	Example: `  faas-cli store deploy figlet
+							faas-cli store deploy figlet
+									--gateway=http://remote-site.com:8080 --lang=python
+									--env=MYVAR=myval`,
+	RunE: runStoreDeploy,
+}
+
+func runStoreList(cmd *cobra.Command, args []string) error {
+	items, err := storeList(storeAddress)
+	if err != nil {
+		return err
+	}
+
+	if len(items) == 0 {
+		fmt.Printf("The store is empty.")
+		return nil
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', 0)
+	fmt.Fprintln(w)
+	fmt.Fprintln(w, "FUNCTION\tDESCRIPTION")
+
+	for _, item := range items {
+		fmt.Fprintf(w, "%s\t%s\n", item.Title, item.Description)
+	}
+
+	fmt.Fprintln(w)
+	w.Flush()
+
+	return nil
+}
+
+func runStoreInspect(cmd *cobra.Command, args []string) error {
+	if len(args) < 1 {
+		return fmt.Errorf("please provide the function name")
+	}
+
+	storeItems, err := storeList(storeAddress)
+	if err != nil {
+		return err
+	}
+
+	item := findFunction(args[0], storeItems)
+	if item == nil {
+		return fmt.Errorf("function '%s' not found", functionName)
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', 0)
+	fmt.Fprintln(w)
+	fmt.Fprintln(w, "FUNCTION\tDESCRIPTION\tIMAGE\tFUNCTION PROCESS\tREPO")
+	fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\n",
```

Comment:
My comment seems to have been removed here by the rebasing. Please use fmt.Sprintf() and build up a string. It should use the spacing directives not tabs. Thank you 👍

Reply:
@alexellis I explained it in another comment that these `\t` are just a separator as I used tabwriter. They're not rendering tab characters. It automatically aligns the columns using spaces.